pandas/plotting/_core.py

```diff
@@ -342,7 +342,9 @@ def _compute_plot_data(self):
                 label = 'None'
             data = data.to_frame(name=label)
 
-        numeric_data = data._convert(datetime=True)._get_numeric_data()
+        # GH16953
+        data = data._convert(datetime=True, timedelta=True)
```

Comment:
I don't think we need the ``_convert`` at all here. this is for object types that need to be coerced. almost all routines in pandas already assume this.

Reply:
The _convert is needed for a test, where a `Series` with numeric values and dtype ``object`` is used (`tests.plotting.test_series.test_valid_object_plot`).
If _convert isn't used, that test will fail in `circleci`.
To prevent the test from failing, _convert can either be used only on `Series` with `dtype==object` and after the `to_frame` (line 344), or on `data` in general after the `Series` specific part (line 346 as is now).